pallets/pallet-asset-switch/README.md

```diff
@@ -0,0 +1,129 @@
+5. `pub fn resume_switch_pair(origin: OriginFor<T>) -> DispatchResult`: Allows the `SwitchOrigin` to resume switches in both directions.
+6. `pub fn update_remote_fee(origin: OriginFor<T>, new: Box<VersionedMultiAsset>) -> DispatchResult`: Allows the `FeeOrigin` to update the required XCM fee to execute the transfer of remote asset on the reserve location from the chain's sovereign account to the beneficiary specified in the `switch` operation. For example, if the cost of sending an XCM message containing a `TransferAsset` instruction from the source chain to AssetHub (reserve location) changes from 0.1 DOTs to 0.2 DOTs, the fee will need to be updated accordingly to avoid transfers to fail on AssetHub, leaving the whole system in an inconsistent state. Since the pallet refunds any unused assets on the reserve location to the account initiating the switch on the source chain, it is not a big issue to overestimate this value here, since no funds will be burnt or unfairly taken from the user during the switch process.
+7. `pub fn switch(origin: OriginFor<T>, local_asset_amount: LocalCurrencyBalanceOf<T, I>, beneficiary: Box<VersionedMultiLocation>) -> DispatchResult`: Allows the `SubmitterOrigin` to perform a switch of some local tokens for the corresponding amount of remote assets on the configured `reserve_location`. The switch will fail already on the source chain if any of the following preconditions is not met:
+	1. The submitter does not have enough balance to pay for the tx fees on the source chain or to cover the amount of local tokens requested. Hence, the user's local balance must be greater than or equal to the amount of tokens requested in the switch + the cost of executing the extrinsic on the source chain.
+	2. No switch pair is set or the switch pair is currently not allowing switches.
+	3. There are not enough locked remote assets on the `reserve_location` to cover the switch request. E.g., if the chain sovereign account on the `reserve_location` only controls `10` remote assets, users can only switch up to `10` local tokens. Once the limit is reached, there needs to be someone performing the reverse operation (remote -> local switch) to free up some remote tokens.
+	4. The switch pair `reserve_location` is not reachable from the source chain, because the configured `XcmRouter` returns an error (e.g., there is no XCM channel between the two chains).
+	5. The configured `SwitchHooks` returns an error in either the `pre-` or the `post-` switch checks.
+	6. The user does not have enough assets to pay for the required remote XCM fees as specified in the switch pair info and as returned by the configured `AssetTransactor`.
+
+## XCM components
+
+Because the switch functionality relies on XCM, the pallet provides a few XCM components that are to be included in a runtime in order to enable the whole set of interactions between the source chain and the configured remote reserve location.
```

Comment:
```suggestion
Because the switch functionality relies on XCM, the pallet provides a few XCM components that should be included in a runtime to enable the whole set of interactions between the source chain and the configured remote reserve location.
```